nuxeo-platform-spreadsheet-web/src/main/js/app/ui/spreadsheet.js

```diff
@@ -57,29 +57,95 @@ class Spreadsheet {
     this.query.depth = 'max';
     // translate directory labels
     this.query.translate('directoryEntry', 'label');
+    this.query.pageProvider = pageProviderName;
 
-    this.query.pageProvider = pageProvider;
-
-    new Layout(connection, layout, language).fetch().then((layout) => {
-      // Check which columns to display
-      var cols = (!columns) ?
-          layout.columns.filter((c) =>  c.selectedByDefault !== false)
-          :
-          columns.map((name) => layout.columns.filter((c) => c.name === name)[0]);
-      this.columns = cols
+    // set columns based on result layout
+    if (resultLayoutName) {
```

Comment:
I know what you mean... needs some refactoring thought, do you consider it a priority?